toolchain/sem_ir/value_stores.h

```diff
@@ -43,6 +43,48 @@ class NodeStore {
   ValueStore<NodeId, Node> values_;
 };
 
+// Provides a ValueStore-like interface for names. This is currently a wrapper
+// around a string store that has no state of its own.
+class NameStore {
+ public:
+  NameStore(ValueStore<StringId>* strings) : strings_(strings) {}
+
+  // Adds a string name, returning an ID to reference it.
+  auto Add(StringId string_id) -> NameId { return NameId(string_id.index); }
+
+  // Returns the requested name as a string ID, if it is a string name. This
+  // returns std::nullopt for special names.
+  auto GetAsStringId(NameId name_id) const -> std::optional<StringId> {
```

Comment:
I'm seeing this called from `GetAsString`, but would it be used otherwise? The extra `std::optional` wrapper versus using `StringId::Invalid` might be a bit of overhead that could be avoided.

Reply:
Thanks, I ended up not needing this for anything. Removed.